lib/plugins/aws/invokeLocal/index.js

```diff
@@ -0,0 +1,110 @@
+'use strict';
+
+const BbPromise = require('bluebird');
+const path = require('path');
+const validate = require('../lib/validate');
+
+class AwsInvokeLocal {
+  constructor(serverless, options) {
+    this.serverless = serverless;
+    this.options = options || {};
+    this.provider = this.serverless.getProvider('aws');
+
+    Object.assign(this, validate);
+
+    this.hooks = {
+      'invoke:local:invoke': () => BbPromise.bind(this)
+        .then(this.extendedValidate)
+        .then(this.invokeLocal),
+    };
+  }
+
+  extendedValidate() {
+    this.validate();
+
+    // validate function exists in service
+    this.options.functionObj = this.serverless.service.getFunction(this.options.function);
+
+    if (this.options.path) {
+      const absolutePath = path.isAbsolute(this.options.path) ?
+        this.options.path :
+        path.join(this.serverless.config.servicePath, this.options.path);
+      if (!this.serverless.utils.fileExistsSync(absolutePath)) {
+        throw new this.serverless.classes.Error('The file you provided does not exist.');
+      }
+      this.options.data = this.serverless.utils.readFileSync(absolutePath);
```

Comment:
Are the options here overwritten only in this class or system wide? Imho this should be done in a local variable and not change the options here